### PR TITLE
kafka(ticdc): sarama admin client use cached topic and partitions

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: ref #8225 

### What is changed and how it works?

* admin client use cached client's topics and partitions

#### Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: pkg/sink/kafka/admin.go [hunk 1]
```diff
@@ -34,11 +34,12 @@ type saramaAdminClient struct {
...
 
 	mu     sync.Mutex
-	client sarama.ClusterAdmin
+	client sarama.Client
+	admin  sarama.ClusterAdmin
 }
 
 const (
-	defaultRetryBackoff  = 20
+	defaultRetryBackoff  = 50
```
Comment (sdojjy) [inline pkg/sink/kafka/admin.go:42]: 50 is too large

**Review by sdojjy — APPROVED** [review]

**Review by zhaoxinyu — APPROVED** [review]

**Pull request merged**